tutor/src/screens/my-courses/dashboard/index.tsx

```diff
@@ -147,16 +150,31 @@ export const MyCoursesDashboard = () => {
     const deleteOffering = () => {
         const tempDisplayedOfferingIds = [...displayedOfferingIds]
         const index = findIndex(tempDisplayedOfferingIds, id => id === deleteOfferingIdModal)
-        if (index >= 0) {
-            tempDisplayedOfferingIds.splice(index, 1)
-            setDisplayedOfferingIds(tempDisplayedOfferingIds)
-            setDeleteOfferingIdModal(null)
+        if (index >= 0 && deleteOfferingIdModal) {
+            const offeringCourses = filter(courses, c => c.offering_id === deleteOfferingIdModal && String(c.term) !== 'preview')
+            Promise.all(map(offeringCourses, c => {
+                const currentRole = find(c.roles, r => r.type === 'teacher')
+                if(currentRole) {
+                    dispatch(dropCourseTeacher(currentRole.id))
+                } else {
+                    Promise.resolve()
+                }
+            }))
+                .then(() => {
+                    tempDisplayedOfferingIds.splice(index, 1)
+                    setDisplayedOfferingIds(tempDisplayedOfferingIds)
+                    setDeleteOfferingIdModal(null)
+                })
         }
     }
 
     const renderDeleteModal = () => {
-        const offeringHasCourses = some(courses, c => c.offering_id === deleteOfferingIdModal)
-        if (!offeringHasCourses) {
+        const offeringCourses = filter(courses, c => c.offering_id === deleteOfferingIdModal)
+        const areAllPreviewCourses = every(offeringCourses, oc => oc.is_preview)
+        const areAllNonPreviewCoursesWithoutStudents = every(filter(offeringCourses, c => String(c.term) !== 'preview'), oc => sumBy(oc?.periods, p => {
+            return p.num_enrolled_students
+        }) === 0)
+        if (areAllPreviewCourses || areAllNonPreviewCoursesWithoutStudents) {
             return (
                 <DeleteOfferingModal
                     show={Boolean(deleteOfferingIdModal)}
```

Comment:
This component has gotten really large.  Can we move any of the functions into their own hook?

Reply:
I agree, I will break it down. thanks nathan!

Reply:
@nathanstitt moved the displayOfferingIds logic into a custom hook. That should shrink the file a little bit.